### PR TITLE
Change regex to a less restrictive one

### PR DESCRIPTION
**Pull request by Pandapip1**
- Allows capital letters in email
- Clearer regex
- More false positives

See https://app.travis-ci.com/github/ethereum/EIPs/jobs/565834359 (the email was `IronchainDAO@gmail.com`)

### CODE REVIEW

File: src/validators.rs [hunk 1]
```diff
@@ -149,7 +149,7 @@ fn validate_author<'a>(acc: &mut Vec<String>, s: &str) -> Result<()> {
             return Err(Error::TrailingInfoAfterEmail);
         }
 
-        let re = Regex::new(r#"(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])"#).unwrap();
+        let re = Regex::new(r#"(?:[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+)"#).unwrap();
```
Comment (MicahZoltu) [inline src/validators.rs:152]: I would prefer gain with the one used in the HTML5 spec: https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address

```suggestion
        let re = Regex::new(r#"^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"#).unwrap();
```

Comment (MicahZoltu) [reply src/validators.rs:152]: At the least, we should ensure we support the full set of allowable and standard characters in the address, like `+`.

Comment (Pandapip1) [reply src/validators.rs:152]: And capital letters :)

**Comment by MicahZoltu** [comment]
Sadly, the Regexp used is for JS and it appears to need some tweaks for Rust.  I bet there is a tool online for converting from JS regexp to rust regexp...

**Pull request closed without merging**